server/questions.js

```diff
@@ -567,3 +568,67 @@ exports.voteReply = function (replyId, vote, userId, callback) {
         }
     });
 }
+
+exports.isUserLocked = function(userId, question, callback){
+    var waiting_time = settings.getQuestionTimeoutPeriod();
+    var lastSubmissionTime;
+    var currentDate = common.getDateObject();
+
+    // check if user already got the question correct
+    for (var obj = 0; obj < question.correctAttempts.length; obj ++){
+        if(question.correctAttempts[obj]['userId'] === userId){
+            return callback(false,null);
+        }
+    }
+    // find the lastSubmissionTime for this user
+    for (var obj = 0; obj < question.userSubmissionTime.length; obj++){
+        if(question.userSubmissionTime[obj]['userId'] === userId){
+            lastSubmissionTime = question.userSubmissionTime[obj]['submissionTime'];
+            break;
+        }
+    }
+
+    if(lastSubmissionTime){
+        const diff = Math.abs(currentDate - lastSubmissionTime);
+        const timeLeftToWait = waiting_time - diff;
+        if (diff < waiting_time){
```

Comment:
You can probably do this in one line, you don't need these extra variables

Reply:
Yes It can be done in one line, however it doesn't tell someone what you are calculating here. I rather not put it in one line

Reply:
I agree with darshan, putting everything in one line will make debugging hell